### PR TITLE
[Merged by Bors] - Add AST node for parenthesized expressions

### PR DESCRIPTION
**Pull request by raskad**
Currently we have no explicit representation for parenthesized expressions which makes some behaviours impossible to detect. A bonus is that we can now turn AST that contains parenthesized expressions back to code.

This Pull Request changes the following:

- Add an AST node for parenthesized expressions.
- Adjust some conversions and checks to "ignore"/"expand" parenthesized expressions.
- Fix some tests that had parenthesized expressions.

### CODE REVIEW

**Review by HalidOdat — APPROVED** [review]
Looks good to me!

File: boa_engine/src/bytecompiler/mod.rs [hunk 1]
```diff
@@ -1239,28 +1245,36 @@ impl<'b, 'host> ByteCompiler<'b, 'host> {
...
+        let (call, mut kind) = match callable {
+            Callable::Call(call) => (call, CallKind::Call),
             Callable::New(new) => (new.call(), CallKind::New),
         };
 
-        match call.function() {
+        let mut function = call.function();
+        while let Expression::Parenthesized(p) = function {
+            function = p.expression();
+        }
```
Comment (jedel1043) [inline boa_engine/src/bytecompiler/mod.rs:1256]: Saw this pattern on several places. Maybe it should be extracted to something like `Expression::flatten`

**Review by jedel1043 — APPROVED** [review]
Just a small nitpick that doesn't block merging. Nice work!

**Pull request closed without merging**